File: tests/fixtures/sample.ml
(* Calculator module *)
module Calculator = struct
  let add x y = x + y

  let multiply x y = x * y

  let factorial n =
    let rec aux acc = function
      | 0 -> acc
      | n -> aux (acc * n) (n - 1)
    in
    aux 1 n
end

(* Variant type for colors *)
type color = Red | Green | Blue

(* Record type for 2D points *)
type point = {
  x : float;
  y : float;
}

(* Type alias *)
type name = string

let greet name =
  Printf.printf "Hello, %s!\n" name

let distance p1 p2 =
  let dx = p1.x -. p2.x in
  let dy = p1.y -. p2.y in
  sqrt (dx *. dx +. dy *. dy)

File: tests/fixtures/sample.jl
# Calculator module
module Calculator

export add, multiply, factorial

function add(x::Int, y::Int)::Int
    return x + y
end

function multiply(x, y)
    return x * y
end

function factorial(n::Int)
    if n <= 1
        return 1
    else
        return n * factorial(n - 1)
    end
end

end # module

# Point struct
struct Point
    x::Float64
    y::Float64
end

# Abstract type
abstract type Shape end

function greet(name::String)
    println("Hello, $name!")
end

function distance(p1::Point, p2::Point)
    dx = p1.x - p2.x
    dy = p1.y - p2.y
    return sqrt(dx^2 + dy^2)
end
